examples/stdlib/test/list_examples_pbt.effekt

```diff
@@ -0,0 +1,166 @@
+import test
+
+def main() = {
+  println(suite("PBT: List Functions and Other Simple Examples") {
+
+    // example unit tests to show that mixed test suites work
+    test("1 + 1 == 2") {
+      assertEqual(1 + 1, 2)
+    }
+
+    test("2 + 2 == 3") {
+      assertEqual(2 + 2, 3)
+    }
+
+    // reverse[x] === [x]
+    with arbitraryInt;
+    forall[Int]("reverse-singleton", 100){ x => 
+      assertEqual(
+        reverse([x]),
+        [x]
+      )
+    }
```

Comment:
When you add the checkfiles, this will probably fail for two reasons on LLVM:
1. we need to use the variant of `forall` that knows how to print the tested value (see previous comments on `forall` defn)
2. `assertEqual` works only on JS; if we want it to work on LLVM, you have to use a more general form of `assertEqual` (defn below), so something like `assertEqual(reverse[x], [x]) { list => <{ /* equality on lists */ }> } { list => show(list) }`
https://github.com/effekt-lang/effekt/blob/55e373b537fcf104419b5ec3685a09e1faa2a5e6/libraries/common/test.effekt#L54 

where you might need to define equality on lists in `libraries/common/list.effekt` if we don't have it already.